nuxeo-runtime/nuxeo-launcher-commons/src/test/java/org/nuxeo/launcher/config/ConfigurationGeneratorTest.java

```diff
@@ -46,16 +47,25 @@
 
 public class ConfigurationGeneratorTest extends AbstractConfigurationTest {
 
+
+    Map<String,String> env = new HashMap<>();
+
     /**
      * @throws java.lang.Exception
      */
     @Override
     @Before
     public void setUp() throws Exception {
         super.setUp();
+        env.put("NUXEO_DB_HOST","10.0.0.1");
         FileUtils.copyDirectory(getResourceFile("templates/jboss"), new File(nuxeoHome, "templates"));
         System.setProperty("jboss.home.dir", nuxeoHome.getPath());
-        configGenerator = new ConfigurationGenerator();
+        configGenerator = new ConfigurationGenerator() {
+            @Override
+            protected String getEnvironmentVariableValue(String key) {
+                return env.get(key);
+            };
+        };
```

Comment:
The test uses `System.getProperty()` but not `System.getEnv()` which is more difficult to emulate in a test class. I don't like creating method only for test purpose but I don't know any other solution to emulate env variable, that's why I added this mock.